src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -85,7 +94,7 @@ void ClusterResourceScheduler::Init(
 }
 
 bool ClusterResourceScheduler::NodeAlive(scheduling::NodeID node_id) const {
-  if (node_id == local_node_id_) {
+  if (!local_node_id_.IsNil() && node_id == local_node_id_) {
```

Comment:
The checking of `!local_node_id_.IsNil()` may be unnecessary?

Reply:
The same reason above.

Reply:
Same reason above

Reply:
Since the `RAY_CHECK(!local_node_id_.IsNil())` will be added inside `Init(...)` (as we discussed offline), the change of this line could be removed?